app/javascript/components/Analytics/SelectPopover.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from "react";
+
+import { Icon } from "$app/components/Icons";
+import { Popover } from "$app/components/Popover";
+
+interface SelectOption<T extends string> {
+  value: T;
+  label: string;
+}
+
+interface SelectPopoverProps<T extends string> {
+  options: SelectOption<T>[];
+  value: T;
+  onChange: (value: T) => void;
+  ariaLabel?: string;
+  size?: "small" | "large";
+}
+
+export function SelectPopover<T extends string>({ options, value, onChange, ariaLabel, size = "small" }: SelectPopoverProps<T>) {
+  const selectedOption = options.find((opt) => opt.value === value);
+  const radioGroupName = React.useId();
+
+  const handleInputChange = (close: () => void) => (event: React.ChangeEvent<HTMLInputElement>) => {
+    const target = event.target as HTMLInputElement;
+    const optionValue = target.value as T;
+    onChange(optionValue);
+    close();
+  };
+
+  const widthClass = size === "small" ? "!w-32" : "!w-40";
+
+  return (
+    <Popover
+      trigger={
+        <span className={`input ${widthClass} flex items-center justify-between whitespace-nowrap`} aria-label={ariaLabel}>
```

Comment:
Please don't interpolate classes direcly and prefer using the `classNames` helper in `$app/utils/classNames`